### PR TITLE
Add build-system to pyproject.toml for pep517 (tox-wheel)

### PR DESCRIPTION
**Pull request by altendky**
Stalled on:
- [ ] https://github.com/tox-dev/tox/issues/1850

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,8 @@
 [tox]
 envlist = cov-erase, flake8, {pypy27,pypy36,pypy37,py27,py35,py36,py37}-{tests,flake8,check-manifest,check-newsfragment}, cov-report
+requires=
+    virtualenv>=20.0.35
+    tox-wheel>=0.5.0
```
Comment (graingert) [inline tox.ini:5]:
```suggestion
    tox-wheel>=0.6.0
isolated_build=true
```

https://tox.readthedocs.io/en/latest/config.html#conf-isolated_build
https://github.com/ionelmc/tox-wheel/pull/6

File: tox.ini [hunk 2]
```diff
@@ -29,7 +32,8 @@ deps =
    coverage
    incremental
    mock
-
+wheel = True
+wheel_pep517 = True
```
Comment (graingert) [inline tox.ini:36]:
```suggestion
```

File: setup.py [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 from setuptools import setup, find_packages
 
+# If incremental is not present then setuptools just silently uses v0.0.0 so
+# let's import it and fail instead.
+import incremental
+
+
 setup(
```
Comment (graingert) [inline setup.py:12]: might be worth doing a `pipx run setup-py-upgrade .` too

Comment (altendky) [reply setup.py:12]: Welp, I was trying to do a quick i-lost-my-momentum-but-still-have-stuff-to-do-but-i-will-release-now-anyways release but...  while we're reworking the project and defining the release process and automating publishing...  `:]`

**Comment by altendky** [comment]
Kick CI

**Comment by altendky** [comment]
@gaborbernat, FYI related to https://github.com/tox-dev/tox/issues/1850, it seems that PyPy itself is unable to create a venv.  Further it does so without a failure exit code from what I can tell.  I noticed this locally first but recreated in CI for sharing.

I would normally put this over in https://github.com/tox-dev/tox/issues/1850 so anyone that looked would be aware, but I can't.  I mention you just in case you were to spend some time on this, I wouldn't want you to do so without being aware of this.  I'll try to dig into this in the PyPy direction later.

https://github.com/twisted/towncrier/pull/321/checks?check_run_id=1995458141
```yaml
    - name: Try creating an env
      if: matrix.python.major == 3
      run: |
        python -m venv a_unique_env_directory
        tree /f a_unique_env_directory
```

```
2021-02-27T21:01:27.3134872Z [36;1mpython -m venv a_unique_env_directory[0m
2021-02-27T21:01:27.3135441Z [36;1mtree /f a_unique_env_directory[0m
2021-02-27T21:01:27.3200612Z shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
2021-02-27T21:01:27.3201167Z env:
2021-02-27T21:01:27.3201713Z   pythonLocation: C:\hostedtoolcache\windows\PyPy\3.7.9\x86
2021-02-27T21:01:27.3202321Z ##[endgroup]
2021-02-27T21:01:28.0784308Z Unable to copy 'C:\\hostedtoolcache\\windows\\PyPy\\3.7.9\\x86\\venvlauncher.exe'
2021-02-27T21:01:28.2212747Z Error: [WinError 2] The system cannot find the file specified
2021-02-27T21:01:28.2590284Z Folder PATH listing for volume Temporary Storage
2021-02-27T21:01:28.2591041Z Volume serial number is 3672-F364
2021-02-27T21:01:28.2591678Z D:\A\TOWNCRIER\TOWNCRIER\A_UNIQUE_ENV_DIRECTORY
2021-02-27T21:01:28.2594289Z �   pyvenv.cfg
2021-02-27T21:01:28.2594893Z �   
2021-02-27T21:01:28.2595728Z ����Include
2021-02-27T21:01:28.2596391Z ����Lib
2021-02-27T21:01:28.2596961Z �   ����site-packages
2021-02-27T21:01:28.2598233Z ����Scripts
2021-02-27T21:01:28.2599258Z         libpypy3-c.dll
2021-02-27T21:01:28.2600784Z         pypy3.exe
```

image

**Comment by gaborbernat** [comment]
Unlocked.

**Comment by altendky** [comment]
Thanks.

For linkage, an updated copy of the above comment is included at https://github.com/tox-dev/tox/issues/1850#issuecomment-787199958.

**Comment by gaborbernat** [comment]
Please open an issue at https://foss.heptapod.net/groups/pypy/-/issues and post back the link here 👍

**Comment by altendky** [comment]
https://github.com/twisted/towncrier/pull/321#discussion_r584541494

@gaborbernat, I believe your report at https://github.com/pypa/virtualenv/issues/2071 is 'in place of' the requested report to PyPy?  Or do you still feel PyPy needs a change?  I want to make sure I don't drop the ball on any of the paths here.

**Comment by gaborbernat** [comment]
I think pypy is here correct, virtualenv is the one mistaken.

**Comment by altendky** [comment]
Went with #323, for now at least.

**Pull request closed without merging**